File: csharp/Creational/02-AbstractFactory/Concrete Factories/BusinessNotificationFactory.cs
public class BusinessNotificationFactory : IClientNotificationFactory
{
    public IAlertNotifier   CreateAlertNotifier()   => new FormalSystemAlert();
    public IEmailReceipt    CreateEmailReceipt()    => new InvoiceEmailReceipt();
    public IDashboardWidget CreateDashboardWidget() => new AdminSummaryWidget();
}

File: typescript/Creational/02-AbstractFactory/Concrete Factories/BusinessNotificationFactory.ts
class BusinessNotificationFactory implements IClientNotificationFactory {
  createAlertNotifier():   IAlertNotifier   { return new FormalSystemAlert(); }
  createEmailReceipt():    IEmailReceipt    { return new InvoiceEmailReceipt(); }
  createDashboardWidget(): IDashboardWidget { return new AdminSummaryWidget(); }
}